erpnext/accounts/doctype/sales_invoice/sales_invoice.py

```
@@ -2097,6 +2099,112 @@ def set_status(self, update=False, status=None, update_modified=True):
+			try:
+				self._recalculate_sales_order_billing(so_name, items_with_discounts)
+			except Exception as e:
+				frappe.log_error(f"Error updating Sales Order {so_name}: {str(e)}", "Sales Order Auto Close")
+
+	def _recalculate_sales_order_billing(self, so_name, invoice_items):
+		"""
+		Recalculate billing percentage for a specific Sales Order using original rates.
+		"""
+		# Get all Sales Order Items for this Sales Order
+		so_items = frappe.get_all(
+			"Sales Order Item", 
```

Ruff (0.12.2)

Trailing whitespace

Remove trailing whitespace

(W291)